dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -324,44 +323,67 @@ protected void githubWebhookDelete(String repository, String gitReference, Strin
             throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
         }
 
-        // Find all workflows and services that are github apps and use the given repo
-        List<Workflow> workflows = workflowDAO.findAllByPath("github.com/" + repository, false).stream().filter(workflow -> Objects.equals(workflow.getMode(), DOCKSTORE_YML)).toList();
+        // List of the IDs of all Workflows that are github apps and use the given repo
+        List<Long> workflowIds = workflowDAO.findAllByPath("github.com/" + repository, false).stream()
+            .filter(workflow -> Objects.equals(workflow.getMode(), DOCKSTORE_YML))
```

Comment:
Mode is an enum, so you could simplify with `==`.

This triggered a memory of `isLegacyVersion`, and got me thinking, but I think it's OK --

* `isLegacyVersion` lets you[ manually refresh versions](https://github.com/dockstore/dockstore/blob/e5b055489d4b811c7928f96be338e54855ca62e1/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/SourceCodeRepoInterface.java#L363) that can normally only be refreshed via the .dockstore.yml flow.
* But it doesn't prevent the deletion/updating of legacy versions via the .dockstore.yml flow, so I think that's OK.